### ISSUE TITLE
MediaInfo disables 'Enable FFmpeg plugin' in Windows

### ISSUE THREAD

**Issue by passat1** [MediaArea/MediaInfo#863]
In latest release (24.05.1) MediaInfo randomly disables 'Enable FFmpeg plugin' in 'Advanced' settings.
After enabling it again it stays enabled for some time and then it just become disabled for no obvious reason.
I notice that again if I open file which has black borders (up and down) but MediaInfo does not detect different 'Active height'.
It is almost always after restarting system which is Windows 10 Enterprise 22H2 with all current updates.

<img width="512" alt="MediaInfo_24 05 1" src="https://github.com/MediaArea/MediaInfo/assets/14957056/74e90d89-f405-43e3-a3ac-74f368503dea">

Also when I have opened such file, go to advanced settings, enable FFmpeg plugin and click 'OK', it opens file info again, but without that info about 'Active height'. So I have to close program and open file again to get it correct.

While writing this message, I opened advanced settings and it was disabled.
I enabled it and twice open file with it, and it was enabled.
Then after ending this message (after few minutes), I checked again, and it was disabled again.
I'm always opening files with MediaInfo using **Windows context menu**.
I can not trust what I see, so every time I open any file, I have to go to advanced settings to see if it's disabled again.

**Comment by g-maxime** [comment]
Is "Enable ffmpeg plugin" the only setting that gets randomly reset?

Does `<User directory>/AppData/Roaming/MediaInfo/Plugin/MediaInfo.cfg` contain `EnableFfmpeg = 1` after the plugin gets disabled?

**Comment by passat1** [comment]
For now it's the only one I've noticed.
I used MediaInfo at least 30 times today, and ffmpeg gets disabled really randomly.

At this moment it is **disabled** again, and in mentioned file is `EnableFfmpeg = 0`.

After I enabled it in settings and after program exit value in file changed again to `EnableFfmpeg = 1`.

**Comment by passat1** [comment]
I had installed 'K-Lite Codec Pack' 18.3.5 Mega on my system, which has included MPC-HC with 'mediainfo.dll' which is not using ffmpeg plugin. I did also check properties with included MediaInfo while watching videos with MPC-HC.
Yesterday I installed update 18.3.6_20240604 which also updated 'mediainfo.dll', and today I spent about an hour checking video files, with and without top/bottom borders.
For now 'Enable FFmpeg plugin' was not disabled at all.
Not only once :)
I will keep checking and report back at evening, or maybe tomorrow morning (CET) after new system boot.

**Comment by JeromeMartinez** [comment]
AFAIK the 2 DLL (ours in `C:\Program Files\MediaInfo` and MPC-HC in its directory) are very separate, weird.

But I reproduce the issue on my machine too, reproductible : launch, open preferences, activate FFmpeg, OK,  open preferences, OK, open preferences, option is deactivated.
Also with older versions so not a new issue from the developments in the latest version.

**Comment by passat1** [comment]
I've found easier way to reproduce:
Launch, open Preferences ('Setup' is shown) then select Advanced, activate FFmpeg and 'OK', select 'Setup', and back 'Advanced'.
FFmpeg is **disabled**.
Meaning, just going from 'Advanced' to 'Setup' and back to 'Advanced' is disabling FFmpeg.

**Comment by passat1** [comment]
Thanks.
I did test snapshot and was **not able** to reproduce problem anyway anymore.
But, if file is opened with **disabled** FFmpeg and then open settings and enable FFmpeg, MediaInfo returns to main screen and displays file info.
Displayed info is without 'Active height' and 'Active display aspect ratio', meaning FFmpeg is **not used**.
Same is in situation if FFmpeg was active before and now is deactivated.
Changes are **not accepted** on return to main screen from settings, and file is not analyzed again with **new settings** until file is not **opened again**.

**Comment by JeromeMartinez** [comment]
> until file is not opened again.

We don't launch again the analysis automatically and we don't plan to change that for the moment (very corner case, and if someone want to implement and test no problem for accepting a patch).
